### ISSUE TITLE
Parse task durations as "1h", "1d" etc

### ISSUE THREAD

**Issue by xy2iii** [tijlleenders/ZinZen-scheduler#18]
Right now, durations are coded as numbers and as 1 hour units:
```
  {
    "id": 2,
    "description": "dentist",
    "task_duration": 2,
    "start": "June 1, 2022 13:00:00",
    "deadline": "June 5, 2022 17:00:00"
  },
```

However, this only works for hours and we'll have to have another field for specifying the time unit.

A solution could be using [RFC3339 durations](https://www.ietf.org/rfc/rfc3339.txt). Durations would look like "3h", "3m" etc.

When looking into this I didn't manage to implement it directly easily, it's pretty involved as time does not support it directly. This may help: https://stackoverflow.com/questions/68611943/how-do-i-parse-iso-8601-duration-strings-using-chrono

**Comment by tijlleenders** [comment]
Yeah, using just a number without a unit (assumimg hours) is on purpose to keep it simple.

The referenced RFC is nice for duration.
'''
dur-second = 1*DIGIT "S" 
dur-minute = 1*DIGIT "M" [dur-second]
dur-hour = 1*DIGIT "H" [dur-minute] 
dur-time = "T" (dur-hour / dur-minute / dur-second) 
dur-day = 1*DIGIT "D" 
dur-week = 1*DIGIT "W" 
dur-month = 1*DIGIT "M" [dur-day] 
dur-year = 1*DIGIT "Y" [dur-month] 
dur-date = (dur-day / dur-month / dur-year) [dur-time] 

duration = "P" (dur-date / dur-time / dur-week)
'''

**Comment by tijlleenders** [comment]
...but I guess we need to relax the 1DIGIT constraint to say 11 hours...

**Comment by thinkrapido** [comment]
In general, I think this is a good idea, but we need to think more of kinda "spans".

Currently, the atomic time slot is a span of 1h.

The new scheduler algorithm #411 takes this concept of span into account and schedules the goals accordin to these spans.

It is a design question how the granular the schedular should be. What makes sense? 1m spans, 15m, or 1h?
At the current state of the new scheduler the algorithm already takes this into account and there should only be on point in code to adjust this.

**Comment by tijlleenders** [comment]
@thinkrapido Agree there should be one place in the code to adjust this!  
I also call this the **granularity** of the scheduler.

I think we're on the same page but using different words for other things:  
- **Slot**: For me this is [start;end[ - respecting the minimum granularity - but can be longer. If not allowed to be longer than 1 unit ... you'll be using a piece of memory for every minute if we ever go down to that level... Ok this is premature optimization - but it is also harder to reason about no?  A Slot can be longer than the Duration required to schedule once.
- **Duration**: The amount of units the Goal needs to be scheduled once.
- **Span**: ... I don't know what you mean by that - the length of a slot?
